### PR TITLE
refactor(halo): implement v0 cosmos app

### PR DESCRIPTION
**Pull request by corverroos**
Add basic `halo2` cosmos app.

Note this doesn't work yet since we need to figure out genesis with proper accounts and balances for validators.

task: none

### CODE REVIEW

**Review by pavelnikolov — APPROVED** [review]
LGTM

File: halo/cmd/cometconfig.go [hunk 1]
```diff
@@ -37,7 +37,8 @@ func DefaultCometConfig(homeDir string) cfg.Config {
 
 	conf.RootDir = homeDir
 	conf.SetRoot(conf.RootDir)
-	conf.LogLevel = "error" // Decrease default comet log level, it is super noisy.
+	conf.LogLevel = "error"                            // Decrease default comet log level, it is super noisy.
+	conf.TxIndex = &cfg.TxIndexConfig{Indexer: "null"} // Disable tx indexing.
```
[the remark this answers is not in the document]
Comment (corverroos) [reply halo/cmd/cometconfig.go:41]: https://docs.cometbft.com/v0.37/app-dev/indexing-transactions

File: halo/cmd/cmd.go [hunk 2]
```diff
@@ -16,18 +17,19 @@ func New() *cobra.Command {
 	return libcmd.NewRootCmd(
 		"halo",
 		"Halo is a consensus client implementation for the Omni Protocol",
-		newRunCmd(app.Run),
+		newRunCmd("run", halo1.Run),
 		newInitCmd(),
+		newRunCmd("run2", halo2.Run),
```
[the remark this answers is not in the document]
Comment (corverroos) [reply halo/cmd/cmd.go:22]: nope

File: halo/app/config.go [hunk 3]
```diff
@@ -26,6 +29,10 @@ const (
 	DefaultHomeDir                 = "./halo" // Defaults to "halo" in current directory
 	defaultAppStatePersistInterval = 1        // Persist app state every block. Set to 0 to disable persistence.
 	defaultSnapshotInterval        = 1000     // Roughly once an hour (given 3s blocks)
+	defaultMinRetainBlocks         = 0        // Retain all blocks
+
+	defaultPruningOption = pruningtypes.PruningOptionNothing // Prune nothing
+	defaultDBBackend     = db.GoLevelDBBackend
```
[the remark this answers is not in the document]
Comment (corverroos) [reply halo/app/config.go:35]: I'll remove the comment

**Pull request merged**